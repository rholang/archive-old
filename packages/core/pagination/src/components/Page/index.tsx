import React, { Component } from 'react';
import Button, { ButtonProps, ButtonAppearances } from '@atlaskit/button';

type Diff<T, U> = T extends U ? never : T;

type PagePropsType = Diff<
  ButtonProps,
  {
    appearance?: ButtonAppearances;
    autoFocus: boolean;
    isDisabled: boolean;
    isLoading: boolean;
    spacing: 'compact' | 'default' | 'none';
    shouldFitContainer: boolean;
    type: 'button' | 'submit';
  }
> & { page?: any };

export default class Page extends Component<PagePropsType> {
  render() {
    return <Button {...this.props} appearance="subtle" />;
  }
}
