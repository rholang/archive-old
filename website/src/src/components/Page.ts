import styled from 'styled-components';
import { gridSize, math } from '@atlaskit/theme';

const PageContainer = styled.main`
  width: 100vw;
  max-width: calc(640px + 4rem);
  box-sizing: border-box;
  margin: 2rem auto;
  padding: 0 2rem;
`;

export default PageContainer;

export const Title = styled.h1`
  margin-bottom: 1em;
`;

export const Section = styled.section`
  margin-top: 3em;

  p {
    line-height: 1.4em;
  }
`;

export const Intro = styled.p`
  font-size: ${math.multiply(gridSize, 2)}px;
  font-weight: 300;
  line-height: 1.4em;
`;
