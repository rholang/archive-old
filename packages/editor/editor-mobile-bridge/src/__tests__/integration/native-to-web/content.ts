import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  callNativeBridge,
  editor,
  editable,
  getDocFromElement,
  navigateOrClear,
  skipBrowsers as skip,
} from '../_utils';

import { invalidAdf } from './__fixtures__/invalid-adf';

BrowserTestCase(
  `Can properly set content when given invalid nodes`,
  { skip },
  async (client: any, testName: string) => {
    const browser = new Page(client);

    await navigateOrClear(browser, editor.path);
    await browser.waitForSelector(`${editable} > p`);
    await callNativeBridge(browser, 'setContent', JSON.stringify(invalidAdf));

    const doc = await browser.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
