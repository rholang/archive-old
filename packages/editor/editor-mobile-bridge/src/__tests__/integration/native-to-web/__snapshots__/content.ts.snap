// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can properly set content when given invalid nodes 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Hello text",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "attrs": Object {
        "originalValue": Object {
          "attrs": Object {},
          "type": "broken",
        },
      },
      "type": "unsupportedBlock",
    },
  ],
  "type": "doc",
}
`;
